### DISCUSSION TITLE
Slow building

### DISCUSSION THREAD

**Discussion by gemiren** [pointfreeco/swift-perception#31] [General]
First of all, thanks for the awesome work. However, I've encountered a performance issue which is becoming more and more unbearable. My project used to finish building in one minute. Now, it takes more than ten minutes to build even after a single line change in a view class. I'm using MacBook Pro 16 2019. Anybody has noticed this extremely long building time?

**Comment by mbrandonw** [comment]
When you say "My project used to finish building in one minute", do you mean before depending on Perception? Or did you have a one minute build time even after adding Perception?

**Reply by gemiren** [reply]
I mean before adding the Perception package to the project.

**Reply by mbrandonw** [reply]
Hi @gemiren, then you are just seeing the cost of using macros in Swift projects. We brought up these concerns in a Swift forums post from many months ago ([here](https://forums.swift.org/t/macro-adoption-concerns-around-swiftsyntax/66588)), and more concerns were brought up recently ([here](https://forums.swift.org/t/compilation-extremely-slow-since-macros-adoption/67921)). There currently is no solution to this problem. If you depend on a project that uses Swift macros, you will incur the cost of building SwiftSyntax, which is hefty.

**Reply by gemiren** [reply]
That's a bummer 😕. Again, thanks for this awesome library.
